components/DiscordBar.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react'
+import styles from '../scss/discordBar.module.scss'
+type Member = {
+  id: string
+  username: string
+  status: string
+  avatar_url: string
+}
+interface discordJSON {
+  presence_count: number
+  members: Member[]
+}
+const ONE_MINUTE = 60 * 1000
+const Member: React.FC<{ avatar: string }> = ({ avatar }) => {
+  return (
+    <div className="d-inline-block">
+      <img src={avatar} className={`${styles['member-avatar']} mx-1`} />
+    </div>
+  )
+}
+const DiscordBar: React.FC = () => {
+  const [data, setData] = useState<discordJSON>({
+    presence_count: 1,
+    members: []
+  })
+  const getData = async () => {
+    try {
+      let data = await fetch(
+        'https://discord.com/api/guilds/828783458469675019/widget.json'
+      )
+      data = await data.json()
+      setData((data as unknown) as discordJSON)
```

Comment:
This way it's better by using a const and without the double type casting.

```suggestion
      const data: discordJSON = await fetch(
        'https://discord.com/api/guilds/828783458469675019/widget.json'
      ).then(r => r.json())
      setData(data)
```

Reply:
I feel like there is no good way to do it. Ideally I'd simple chain promises but I'm forced to use `try/catch` for consistency. I removed ugly typescript casting by adding additional variable but it doesn't feel right either way.